.github/workflows/require-pr-label.yml

```
@@ -0,0 +1,16 @@
+name: Require PR label
```

Hmm, what about "do not merge" or "merging is blocked"?

---

IMO, since both the workflow name and the job name is shown in the checks list, for clarity without redundancy I suggest 

```suggestion
name: Check labels
```

here which, along with `name: DO-NOT-MERGE` below, would render as `Check labels/DO-NOT-MERGE`.

This also scales if we add more label checks or similar, particularly by other successor actions to Bedevere.